httpcore/_backends/curio.py

```diff
@@ -0,0 +1,214 @@
+import select
+import socket
+from ssl import SSLContext, SSLSocket
+from typing import Dict, Optional, Type, Union
+
+import curio
+import curio.io
+
+from .._exceptions import (
+    ConnectError,
+    ConnectTimeout,
+    ReadError,
+    ReadTimeout,
+    TimeoutException,
+    WriteError,
+    WriteTimeout,
+    map_exceptions,
+)
+from .._types import TimeoutDict
+from .._utils import get_logger
+from .base import AsyncBackend, AsyncLock, AsyncSemaphore, AsyncSocketStream
+
+logger = get_logger("curio_backend")
+
+one_day_in_seconds = 60 * 60 * 24
+
+
+async def wrap_ssl_client(
+    sock: curio.io.Socket,
+    ssl_context: SSLContext,
+    server_hostname: bytes,
+) -> curio.io.Socket:
+    kwargs = {
+        "server_hostname": server_hostname,
+        "do_handshake_on_connect": sock._socket.gettimeout() != 0.0,
+    }
+
+    socket = curio.io.Socket(ssl_context.wrap_socket(sock._socket, **kwargs))
```

Comment:
(Nit) The `**kwargs` style is a bit surprising, should we consider passing kwargs directly to `.wrap_socket()`?

```suggestion
    socket = curio.io.Socket(
        ssl_context.wrap_socket(
            sock._socket,   
            server_hostname=server_hostname,
            do_handshake_on_connect=sock._socket.gettimeout() != 0,
        )
    )
```

Reply:
Yes, kwargs wasn't a good idea here :-) changed